framework/decode/dx12_replay_consumer_base.cpp

```diff
@@ -3771,5 +3801,577 @@ std::wstring Dx12ReplayConsumerBase::ConstructObjectName(format::HandleId captur
+    StructPointerDecoder<Decoded_D3D12_CPU_DESCRIPTOR_HANDLE>* pRenderTargetDescriptors,
+    BOOL                                                       RTsSingleHandleToDescriptorRange,
+    StructPointerDecoder<Decoded_D3D12_CPU_DESCRIPTOR_HANDLE>* pDepthStencilDescriptor)
+{
+    if (options_.dump_resources_type != DumpResourcesType::kNone &&
+        track_dump_resources_.target.set_render_targets_code_index == call_info.index)
+    {
+        auto rt_handles = pRenderTargetDescriptors->GetMetaStructPointer();
+        track_dump_resources_.render_target_heap_ids.resize(NumRenderTargetDescriptors);
+        track_dump_resources_.replay_render_target_handles.resize(NumRenderTargetDescriptors);
+        for (uint32_t i = 0; i < NumRenderTargetDescriptors; ++i)
+        {
+            track_dump_resources_.render_target_heap_ids[i]       = rt_handles[i].heap_id;
+            track_dump_resources_.replay_render_target_handles[i] = *rt_handles[i].decoded_value;
+        }
+        auto ds_handle = pDepthStencilDescriptor->GetMetaStructPointer();
+        if (ds_handle)
+        {
+            track_dump_resources_.depth_stencil_heap_id       = ds_handle->heap_id;
+            track_dump_resources_.replay_depth_stencil_handle = *ds_handle->decoded_value;
+        }
+    }
+}
+
+void Dx12ReplayConsumerBase::PreCall_ID3D12GraphicsCommandList4_BeginRenderPass(
+    const ApiCallInfo&                                                  call_info,
+    DxObjectInfo*                                                       object_info,
+    UINT                                                                NumRenderTargets,
+    StructPointerDecoder<Decoded_D3D12_RENDER_PASS_RENDER_TARGET_DESC>* pRenderTargets,
+    StructPointerDecoder<Decoded_D3D12_RENDER_PASS_DEPTH_STENCIL_DESC>* pDepthStencil,
+    D3D12_RENDER_PASS_FLAGS                                             Flags)
+{
+    if (options_.dump_resources_type != DumpResourcesType::kNone &&
+        call_info.index == track_dump_resources_.target.begin_renderpass_code_index)
+    {
+        // This should be in the override function since it modifies the parameters. But the override function
+        // doesn't have call_info, so stay here.
+
+        // Since it will insert renderpass, this EndingAccess has to be PRESERVE.
+        // And then the inserted renderpass's BeginningAccess has to be PRESERVE.
+        auto rt_descs = pRenderTargets->GetMetaStructPointer();
+        for (uint32_t i = 0; i < NumRenderTargets; ++i)
+        {
+            rt_descs[i].EndingAccess->decoded_value->Type = D3D12_RENDER_PASS_ENDING_ACCESS_TYPE_PRESERVE;
+        }
+
+        auto ds_desc = pDepthStencil->GetMetaStructPointer();
+        if (ds_desc)
+        {
+            ds_desc->DepthEndingAccess->decoded_value->Type   = D3D12_RENDER_PASS_ENDING_ACCESS_TYPE_PRESERVE;
+            ds_desc->StencilEndingAccess->decoded_value->Type = D3D12_RENDER_PASS_ENDING_ACCESS_TYPE_PRESERVE;
```

Comment:
The EndingAcccess of the renderpass and the BeginningAccess of following renderpasses have to be PRESERVE to keep the the render process for this following renderpass, or it will clear the render target.